semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -354,6 +354,7 @@ message LiteralType {
     DOUBLE = 9;
     STRING = 10;
     NULL = 11;
+    ENUM = 12;
```

Comment:
Let's reflect on whether we should copy/paste what Scalac does here. 

To me, [enum constants](https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.9.1), e.g. `NOFOLLOW_LINKS` look pretty similar to objects in that they are singletons and therefore should be modelled with `SingleType`.

Now, why didn't Scalac go the same route? I wasn't around when this functionality was introduced. However, I have an educated guess. 

I think it's an unfortunate side-effect of the split between `StaticAnnotation` and `ClassfileAnnotation`. The latter only supports `javaArgs` (/cc @MasseGuillaume) that only accept `Constant`, nested annotations with `javaArgs` and arrays thereof. Now, because the JLS allows enum constants in annotations and because Scala must interoperate with Java, enum constants were stuffed into `Constant` for the simplest possible implementation. This decision to prioritize implementation simplicity looks consistent with a number of other hacks in other compiler modules that were introduced to support annotations.

Clearly we don't have the same constraints as Scalac developers had, which is why I propose we do the following, which seems to be the cleanest conceptually:
  * Add `SymbolInformation.ENUM` and `SymbolInformation.ENUM_CONSTANT`.
  * Model `LiteralType(Constant(sym: Symbol))` with `SingleType(sym.owner.thisPrefix, sym)`.
  * Update the specification for `LiteralType` and `SingleType`.

The test case from below will become:
```
_root_.java.nio.file.LinkOption. => final enum LinkOption
_root_.java.nio.file.LinkOption.NOFOLLOW_LINKS. => final enum constant NOFOLLOW_LINKS: NOFOLLOW_LINKS.type
  NOFOLLOW_LINKS => _root_.java.nio.file.LinkOption.NOFOLLOW_LINKS.
```